pkg/pillar/hypervisor/qmp.go

```diff
@@ -66,3 +68,56 @@ func getQemuStatus(socket string) (string, error) {
 		return "", err
 	}
 }
+
+//quitStoppedContainer : Since on shutdown event, qemu process will be stopped but not removed,
+//this method waits (10 min max) for the domain to reach shutdown state and then quit the domain.
+func quitStoppedContainer(socket string) {
+	//NOTE: Since only one connection can be active on a qmp socket, if we keep the connection open until this method exists,
+	// it'll lead to a deadlock as this routine which holds the connection will be waiting for the domain state to change
+	// and the call which changes the state will wait for this routine to disconnect the monitor.
+	//Hence we should open a connection, check for the state and close it immediately
+
+	//Try for 10 min (300 times * 2 sec tick) and then give up as we will eventually clean-up this domain.
+	maxNumberOfTicks := 300
```

Comment:
Hey @adarsh-zededa -- can you please check if you can avoid all this complexity by simply blocking on a call to `{ 'execute': 'system_powerdown' }` and then essentially expecting `SHUTDOWN` to come back to you?

I believe that'll allow you not to worry about polling and also eliminate the timeout -- since it is either that you'll receive your SHUTDOWN or domain will get really killed after 10min and your blocking read will exit anyway.

Reply:
@rvs I tried that but executing `{ 'execute': 'system_powerdown' }` returned `{"return": {}}`. So to see it we reached shutdown state I had to reconnect and listen to events which we can't do for very long. 
Tried another approach:
I created another socket dedicated just for listening events form qmp. I started a routine as soon as we create the domain and the routine will be running until the domain is alive. The routine keeps listening for events and acts accordingly. Also, since that routine holds the connection for the listening port, we will not be able to connect to that port from anywhere else in the code.